src/portals/InputCapture.cpp

```diff
@@ -0,0 +1,594 @@
+
+    Debug::log(LOG, "[input-capture]  | {}", sessionHandle.c_str());
+    Debug::log(LOG, "[input-capture]  | appid: {}", appID);
+
+    if (!sessionValid(sessionHandle))
+        return complete(call);
+
+    disable(sessionHandle);
+
+    complete(call);
+}
+
+void CInputCapturePortal::onEnable(sdbus::MethodCall& call) {
+    Debug::log(LOG, "[input-capture] New Enable request:");
+
+    sdbus::ObjectPath sessionHandle;
+    call >> sessionHandle;
+
+    std::string appID;
+    call >> appID;
+
+    Debug::log(LOG, "[input-capture]  | {}", sessionHandle.c_str());
+    Debug::log(LOG, "[input-capture]  | appid: {}", appID);
+
+    if (!sessionValid(sessionHandle))
+        return complete(call);
+
+    sessions[sessionHandle]->status = ENABLED;
+
+    complete(call);
+}
+
+void CInputCapturePortal::onRelease(sdbus::MethodCall& call) {
+    Debug::log(LOG, "[input-capture] New Release request:");
+
+    sdbus::ObjectPath sessionHandle;
+    call >> sessionHandle;
+
+    std::string appID;
+    call >> appID;
+
+    Debug::log(LOG, "[input-capture]  | {}", sessionHandle.c_str());
+    Debug::log(LOG, "[input-capture]  | appid: {}", appID);
+
+    if (!sessionValid(sessionHandle))
+        return complete(call);
+
+    std::unordered_map<std::string, sdbus::Variant> options;
+    call >> options;
+    uint32_t activationId = options["activation_id"];
+    if (activationId != sessions[sessionHandle]->activationId) {
+        Debug::log(WARN, "[input-capture] Invalid activation id {} expected {}", activationId, sessions[sessionHandle]->activationId);
+        complete(call);
+        return;
+    }
+
+    deactivate(sessionHandle);
+
+    //TODO: maybe warp pointer
+
+    complete(call);
+}
+
+void CInputCapturePortal::onConnectToEIS(sdbus::MethodCall& call) {
+    Debug::log(LOG, "[input-capture] New ConnectToEIS request:");
+
+    sdbus::ObjectPath sessionHandle;
+    call >> sessionHandle;
+
+    std::string appID;
+    call >> appID;
+
+    std::unordered_map<std::string, sdbus::Variant> options;
+    call >> options;
+
+    Debug::log(LOG, "[input-capture]  | {}", sessionHandle.c_str());
+    Debug::log(LOG, "[input-capture]  | appid: {}", appID);
+
+    if (!sessionValid(sessionHandle))
+        return complete(call);
+
+    int sockfd = sessions[sessionHandle]->eis->getFileDescriptor();
+
+    Debug::log(LOG, "[input-capture] Connected to the socket. File descriptor: {}", sockfd);
+    auto reply = call.createReply();
+    reply << (sdbus::UnixFd)sockfd;
+    reply.send();
+}
+
+bool CInputCapturePortal::sessionValid(sdbus::ObjectPath sessionHandle) {
+    if (!sessions.contains(sessionHandle)) {
+        Debug::log(WARN, "[input-capture] Unknown session handle: {}", sessionHandle.c_str());
+        return false;
+    }
+
+    return sessions[sessionHandle]->status != STOPPED;
+}
+
+bool get_line_intersection(double p0_x, double p0_y, double p1_x, double p1_y, double p2_x, double p2_y, double p3_x, double p3_y, double* i_x, double* i_y) {
```

Comment:
no snake

Reply:
Kinda wont fix, i will toss it and write a thing that is more in line with the specification of the input capture portal.